src/mapgen/mg_decoration.cpp

```diff
@@ -472,3 +473,27 @@ size_t DecoSchematic::generate(MMVManip *vm, PcgRandom *pr, v3s16 p, bool ceilin
 
 	return 1;
 }
+
+///////////////////////////////////////////////////////////////////////////////
+ObjDef *DecoLSystem::clone() const
+{
+	auto def = new DecoLSystem();
+	Decoration::cloneTo(def);
+
+	def->tree_def = tree_def;
+	return def;
+}
+
+
+size_t DecoLSystem::generate(MMVManip *vm, PcgRandom *pr, v3s16 p, bool ceiling)
+{
+	if (!canPlaceDecoration(vm, p))
+		return 0;
+	return treegen::make_ltree(*vm, p, m_ndef, *tree_def);
```

Comment:
to make sure it's definitely not used as non-const I'd do this:
```suggestion
	// shared
	const auto &ref = *tree_def;
	return treegen::make_ltree(*vm, p, m_ndef, ref);
```